projects/Weblate/tests/integration-test.nix

```diff
@@ -0,0 +1,123 @@
+{
+  sources,
+  pkgs,
+  ...
+}: let
+  certs = import "${sources.nixpkgs}/nixos/tests/common/acme/server/snakeoil-certs.nix";
+
+  serverDomain = certs.domain;
+
+  admin = {
+    username = "admin";
+    password = "snakeoilpass";
+  };
+  # An API token that we manually insert into the db as a valid one.
+  apiToken = "OVJh65sXaAfQMZ4NTcIGbFZIyBZbEZqWTi7azdDf";
+in {
+  name = "weblate";
+  meta.maintainers = with pkgs.lib.maintainers; [erictapen];
+
+  nodes.server = {
+    config,
+    pkgs,
+    lib,
+    ...
+  }: {
+    virtualisation.memorySize = 2048;
+
+    services.postgresql.package = pkgs.postgresql_14;
+
+    imports = [sources.modules."services.weblate"];
+
+    services.weblate = {
+      enable = true;
+      localDomain = "${serverDomain}";
+      djangoSecretKeyFile = pkgs.writeText "weblate-django-secret" "thisissnakeoilsecret";
+      smtp = {
+        createLocally = true;
+        user = "weblate@${serverDomain}";
+        passwordFile = pkgs.writeText "weblate-smtp-pass" "thisissnakeoilpassword";
+      };
+    };
+
+    services.nginx.virtualHosts."${serverDomain}" = {
+      enableACME = lib.mkForce false;
+      sslCertificate = certs."${serverDomain}".cert;
+      sslCertificateKey = certs."${serverDomain}".key;
+    };
+
+    services.postfix = {
+      enableSubmission = true;
+      enableSubmissions = true;
+      submissionsOptions = {
+        smtpd_sasl_auth_enable = "yes";
+        smtpd_client_restrictions = "permit";
+      };
+      # sslKey = certs.${serverDomain}.key;
+      # sslCert = certs.${serverDomain}.cert;
+    };
+
+    security.pki.certificateFiles = [certs.ca.cert];
+
+    networking.hosts."::1" = ["${serverDomain}"];
+    networking.firewall.allowedTCPPorts = [80 443];
+
+    # We need weblate-env available to the root user.
+    environment.systemPackages = config.users.users.weblate.packages;
+    users.users.weblate.shell = pkgs.bashInteractive;
```

Comment:
Could you maybe pull out everything that's needed to self-host this into an example that will be listed in the overview? https://ngi-nix.github.io/ngipkgs/

Reply:
Added a very basic but working one, is that enough? It's essentially what I have running at https://weblate.erictapen.name/ for testing.

Reply:
But that's not being tested, right? Users still have to somehow know all the other things that would go into a working setup, essentially everything in this test case here.

Reply:
No, everything else in the `server` configuration is purely there to make it work in the test environment. Big chunk is due to the absence of a publicly reachable Letsencrypt infrastructure (as we are in a sandbox).